pkg/controller/statusmanager/status_manager.go

```diff
@@ -114,6 +207,8 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...confi
 
 		oldStatus := co.Status.DeepCopy()
 		status.deleteRelatedObjectsNotRendered(co)
+		status.deleteDaemonSetNotRendered(co)
+		status.deleteDeploymentNotRendered(co)
```

Comment:
why is this needed? The DaemonSets and Deployments should also be in RelatedObjects

Reply:
The origin intention is to remove the deployment and daemonset before other objects. But I found it is unnecessary after more tests. It shall be removed.